core/src/main/java/eu/fasten/core/maven/data/Exclusion.java

```diff
@@ -17,47 +17,73 @@
 
 import java.io.Serializable;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-
 public class Exclusion implements Serializable {
 
-	private static final long serialVersionUID = -1350444195222504726L;
+    private static final long serialVersionUID = -1350444195222504726L;
+
+    public final String artifactId;
+    public final String groupId;
+    private final int hashCode;
+
+    public Exclusion(String groupId, String artifactId) {
+        this.groupId = Ids.gid(groupId);
+        this.artifactId = Ids.aid(artifactId);
+        this.hashCode = calcHashCode();
+    }
 
-	public String artifactId;
-	public String groupId;
+    public int calcHashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((artifactId == null) ? 0 : artifactId.hashCode());
+        result = prime * result + ((groupId == null) ? 0 : groupId.hashCode());
+        return result;
+    }
+
+    @Override
+    public int hashCode() {
+        return hashCode;
+    }
 
-    @SuppressWarnings("unused")
-    private Exclusion() {
-        // exists for JSON object mappers
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Exclusion other = (Exclusion) obj;
+        if (hashCode != other.hashCode) {
+            return false;
+        }
+        if (artifactId == null) {
+            if (other.artifactId != null)
+                return false;
+        } else if (!artifactId.equals(other.artifactId))
+            return false;
+        if (groupId == null) {
+            if (other.groupId != null)
+                return false;
+        } else if (!groupId.equals(other.groupId))
+            return false;
+        return true;
     }
 
-	public Exclusion(final String groupId, final String artifactId) {
-		this.groupId = groupId;
-		this.artifactId = artifactId;
-	}
-
-	public String toJSON() {
-		return String.format("%s:%s", groupId, artifactId);
-	}
-
-	@Override
-	public boolean equals(Object obj) {
-		return EqualsBuilder.reflectionEquals(this, obj);
-	}
-
-	@Override
-	public int hashCode() {
-		return HashCodeBuilder.reflectionHashCode(this);
-	}
-
-	@Override
-	public String toString() {
-		return toJSON();
-	}
-
-	public static Exclusion fromJSON(String json) {
-		String[] parts = json.split(":");
-		return new Exclusion(parts[0], parts[1]);
-	}
+    @Override
+    public String toString() {
+        return String.format("%s:%s", groupId, artifactId);
+    }
+
+    // TODO remove everything below
```

Comment:
What about this one?

Reply:
This is just a TODO for the future. At some point, the remaining code in that class should be removed by refactoring the JSON handling. This is out-of-scope for this PR though.